### PR TITLE
Aro pvt udr

### PR DESCRIPTION
**Pull request by mgahagan73**
Install ARO private cluster with UserDefinedRouting

### CODE REVIEW

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-aro

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-aro

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-aro

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-aro-4.14-aro-f28

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-aro-4.14-aro-f28

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-aro-4.14-aro-f28

**Comment by mgahagan73** [comment]
This PR builds a private ARO cluster using the existing bastion host chain so it should work with existing tests for private clusters. The address space for the aro vnet had to be expanded to accomidate the subnet and nsg the bastion host uses so we don't have to re-write the bastion chain just for ARO. Some hard-coded values changed to variables for future cluster profile installs as well as fixing the ARO versions so the demo rehearse job will continue to work .

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/pvt/cucushift-installer-rehearse-azure-aro-pvt-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+workflow:
+  as: cucushift-installer-rehearse-azure-aro-pvt
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/pvt/cucushift-installer-rehearse-azure-aro-pvt-workflow.yaml:2]: `pvt` means `private`, right? If yes, suggest replace `pvt` with `private` full name.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/pvt/cucushift-installer-rehearse-azure-aro-pvt-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    - ref: azure-provision-resourcegroup
+    - ref: aro-provision-vnet
+    - chain: azure-provision-bastionhost
+    - ref: proxy-config-generate
+    - ref: aro-provision-cluster
+    - ref: ipi-install-rbac
+    - ref: openshift-cluster-bot-rbac
+    - ref: enable-qe-catalogsource
+    post:
+    - ref: aro-deprovision
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/aro/pvt/cucushift-installer-rehearse-azure-aro-pvt-workflow.yaml:14]: If this workflow is used for private cluster install, suggest to set the default env vars here.
```
    env:
      ARO_API_SERVER_VISIBILITY: Private
      ARO_INGRESS_VISIBILITY: Private
      ARO_OUTBOUND_TYPE: UserDefinedRouting
```

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-aro periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-aro-4.14-aro-f28

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-aro-private

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehears-azure-aro-private

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehears-azure-aro-private

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.13.yaml [hunk 3]
```diff
@@ -69,8 +69,15 @@ tests:
...
-      ARO_CLUSTER_VERSION: 4.11.26
+      ARO_CLUSTER_VERSION: 4.13.40
     workflow: cucushift-installer-rehearse-azure-aro
+- as: installer-rehears-azure-aro-private
+  cron: '@yearly'
+  steps:
+    cluster_profile: azure-qe
+    env:
+      ARO_CLUSTER_VERSION: 4.13.40
+    workflow: cucushift-installer-rehearse-azure-aro-private
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.13.yaml:80]: Just one minor comment, suggest to replace the workflow at line 73 with this one to save one more job.

**Comment by jianlinliu** [comment]
Rehearse [job](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55887/rehearse-55887-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehears-azure-aro-private/1829691913108721664) failed.

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehears-azure-aro-private

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehears-azure-aro-private

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehears-azure-aro-private

**Comment by mgahagan73** [comment]
job failures are caused by the az-cli being too old, trying to see if we have a upi-installer image with a newer az cli or if I'll have to make one.

**Comment by jianlinliu** [comment]
> job failures are caused by the az-cli being too old, trying to see if we have a upi-installer image with a newer az cli or if I'll have to make one.

```
$ cat ci-operator/step-registry/aro/provision/cluster/aro-provision-cluster-ref.yaml
ref:
  as: aro-provision-cluster
  from_image:
    namespace: ocp
    name: "4.14"
    tag: upi-installer
  commands: aro-provision-cluster-commands.sh
```
---> https://github.com/openshift/installer/blob/release-4.14/images/installer/Dockerfile.upi.ci.rhel8#L30
```
RUN yum update -y && \
    yum install --setopt=tsflags=nodocs -y \
      azure-cli-2.49.0-1.el8 \
```

You might need to change `aro-provision-cluster` ref to use 4.17 upi-installer, because it is installing the latest azure-cli, refer to https://github.com/openshift-splat-team/pwsh/blob/main/install-azure-cli.sh

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehears-azure-aro-private

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehears-azure-aro-private

**Comment by mgahagan73** [comment]
looks like this is working now, had to add the proxy as well. ARO just released 4.15 so I hope it's ok I left the 4.15 job around.

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehears-azure-aro-private

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.13.yaml [hunk 4]
```diff
@@ -64,13 +64,13 @@ tests:
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
     workflow: cucushift-installer-rehearse-azure-ipi-sdn
-- as: installer-rehearse-azure-aro
+- as: installer-rehears-azure-aro-private
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.13.yaml:67]: I do not think we need to change the job name here.

**Comment by jianlinliu** [comment]
BTW, pls squash multiple commits into one.

File: ci-operator/jobs/coreos/coreos-assembler/coreos-coreos-assembler-rhcos-4.13-presubmits.yaml [hunk 5]
```diff
@@ -60,7 +60,7 @@ presubmits:
     branches:
     - ^rhcos-4\.13$
     - ^rhcos-4\.13-
-    cluster: build03
```
Comment (jianlinliu) [inline ci-operator/jobs/coreos/coreos-assembler/coreos-coreos-assembler-rhcos-4.13-presubmits.yaml:63]: A rebase issue ?

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.13.yaml [hunk 6]
```diff
@@ -64,13 +64,13 @@ tests:
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
     workflow: cucushift-installer-rehearse-azure-ipi-sdn
-- as: installer-rehearse-azure-aro
+- as: installer-rehears-azure-aro
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.13.yaml:67]:
```suggestion
- as: installer-rehearse-azure-aro
```
A typo ?

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.15.yaml [hunk 7]
```diff
@@ -66,6 +66,13 @@ tests:
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
     workflow: cucushift-installer-rehearse-azure-ipi-disk-encryption-set-day2
+- as: installer-rehears-azure-aro-private
```
Comment (jianlinliu) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installer-rehearse-4.15.yaml:69]:
```suggestion
- as: installer-rehearse-azure-aro
```

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Pull request merged**